### ISSUE TITLE
(cli): unable to hotswap when adding tags to lambda function

### ISSUE THREAD

**Issue by cecheta-lego** [aws/aws-cdk#21556]
### Describe the bug

The CDK crashes when attempting to add tags to a lambda function while using hotswap.

### Expected Behavior

The tags are added to the lambda function.

### Current Behavior

```
 ❌  TestStack failed: TypeError: Cannot read properties of undefined (reading 'forEach')
    at isLambdaFunctionCodeOnlyChange (/Users/test/node_modules/aws-cdk/lib/api/hotswap/lambda-functions.ts:169:32)
    at isHotswappableLambdaFunctionChange (/Users/test/node_modules/aws-cdk/lib/api/hotswap/lambda-functions.ts:34:34)
    at findAllHotswappableChanges (/Users/test/node_modules/aws-cdk/lib/api/hotswap-deployments.ts:96:9)
    at tryHotswapDeployment (/Users/test/node_modules/aws-cdk/lib/api/hotswap-deployments.ts:50:37)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at deployStack (/Users/test/node_modules/aws-cdk/lib/api/deploy-stack.ts:274:39)
    at CdkToolkit.deploy (/Users/test/node_modules/aws-cdk/lib/cdk-toolkit.ts:210:24)
    at initCommandLine (/Users/test/node_modules/aws-cdk/lib/cli.ts:346:12)

Cannot read properties of undefined (reading 'forEach')
```

### Reproduction Steps

* Create lambda function, deploy stack

```ts
const func = new lambda.Function(this, "Function", {
  code: lambda.Code.fromInline(`exports.handler = async () => {}`),
  handler: "index.handler",
  runtime: lambda.Runtime.NODEJS_16_X,
});
```

* Add tags, deploy stack

```ts
Tags.of(func).add("key", "value");
```

`cdk deploy --hotswap`

### CDK CLI Version

2.37.1

### Node.js Version

16.16.0

### OS

Mac OS

### Language

Typescript

### Language Version

3.9.7

**Comment by comcalvi** [comment]
yep, this is bad. It turns out that hotswapping tags is entirely broken, and they can't be updated in any way today through hotswap (creating / destroying the `Tags` prop completely causes this crash, and doing additions / deletions of tags that already have a `Tags` prop causes a full deployment). 

Question: Why do you need to hotswap tags? Why is this something that needs to be done each iteration cycle?

**Comment by comcalvi** [comment]
falling back to a full deployment is the intended behavior. There seems to be no use case for the tags themselves to need to be hotswapped, so we're considering removing the Tag hotswap ability (it doesn't work anyway) and ignoring non-hotswappable property updates in a new flag.

**Comment by comcalvi** [comment]
closed by #23653
